dex/src/main/scala/com/wavesplatform/dex/queue/LocalMatcherQueue.scala

```diff
@@ -112,8 +113,9 @@ object LocalMatcherQueue {
       executor.submit(new Runnable {
         override def run(): Unit = {
           val ts = time.correctedTime()
-          store.enqueue(command, time.correctedTime()).foreach { offset =>
-            p.success(ValidatedCommandWithMeta(offset, ts, command))
+          store.enqueue(command, time.correctedTime()).onComplete {
+            case Success(offset) => p.success(ValidatedCommandWithMeta(offset, ts, command))
+            case Failure(ex) => log.error(s"Got exception ${ex.getWithStackTrace} when enqueueing command $command")
           }
```

Comment:
Sorry, we have `Promise` there. It'd better to use `completeWith`. Then the error will be handled futher.

Reply:
yeah, it looks better and it will be handled by CircuitBreaker in this case. I fixed it